recipe/patches/0013-Adjust-CUDAHooks-hasMAGMA-for-libtorch_cuda_linalg.s.patch

```diff
@@ -0,0 +1,53 @@
+From aba2dbc13211d44a2d75dd3ca3084c75327a5903 Mon Sep 17 00:00:00 2001
+From: =?UTF-8?q?Micha=C5=82=20G=C3=B3rny?= <mgorny@gentoo.org>
+Date: Mon, 16 Dec 2024 13:50:44 +0100
+Subject: [PATCH] Adjust CUDAHooks::hasMAGMA() for libtorch_cuda_linalg.so
+ replacement in conda-forge
+
+Update CUDAHooks::hasMAGMA() to detect whether the currently installed
+version of `libtorch_cuda_linalg.so` has been linked against magma or
+not.
+---
+ aten/src/ATen/cuda/detail/CUDAHooks.cpp | 22 ++++++++++++++++++++++
+ 1 file changed, 22 insertions(+)
+
+diff --git a/aten/src/ATen/cuda/detail/CUDAHooks.cpp b/aten/src/ATen/cuda/detail/CUDAHooks.cpp
+index c6e83fad1..a370c2580 100644
+--- a/aten/src/ATen/cuda/detail/CUDAHooks.cpp
++++ b/aten/src/ATen/cuda/detail/CUDAHooks.cpp
+@@ -1,3 +1,10 @@
++#if defined(BUILD_LAZY_CUDA_LINALG)
++#define _GNU_SOURCE
++#include <dlfcn.h>
++#include <libgen.h>
++#include <cstdio>
++#endif
++
+ #include <ATen/cuda/detail/CUDAHooks.h>
+ 
+ #include <ATen/cuda/CUDAGeneratorImpl.h>
+@@ -159,6 +166,21 @@ bool CUDAHooks::hasCUDA() const {
+ }
+ 
+ bool CUDAHooks::hasMAGMA() const {
++#if defined(BUILD_LAZY_CUDA_LINALG)
++  // Conda patch: we support swapping magma-enabled libtroch_cuda_linalg.so
++  // for non-magma-enabled.  The latter cannot be installed along with
++  // libmagma.so, so check for its presence to determine which one is
++  // installed.
++  Dl_info info;
++  if (dladdr(CUDAHooks::hasMAGMA, &info)) {
++    std::string path = info.dli_fname;
++    char* directory = dirname(path.data());
++    path = directory;
++    path += "/libmagma.so";
++    printf("%s\n", path.c_str());
++  }
++#endif
```

Comment:
This doesn't return anything right?

Reply:
Ah, sorry, forgot to `git commit` the non-debug version of the patch :facepalm:.

Reply:
Fixed now.